0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+    "account": "http://usd-ledger.example/accounts/{name}",
+    "websocket": "wss://usd-ledger.example/websocket/"
+  },
+  "precision": 10,
+  "scale": 2
+}
+```
+
+
+#### Errors
+
+- This method does not return any errors under normal circumstances.
+
+
+### Prepare Transfer
+[Prepare Transfer]: #prepare-transfer
+
+Prepares a new transfer (conditional or unconditional) in the ledger. When a transfer becomes prepared, it executes immediately if there is no condition. If you specify `execution_condition`, the funds are held until the beneficiary [submits the matching fulfillment][Submit Fulfillment] or the `expires_at` time is reached.
+
+**Authorization:** The owner of an account MUST be able to prepare a conditional transfer that debits the account. Non-administrators MUST NOT be able to prepare a transfer that debits a different account. A ledger MAY disallow non-administrators from preparing unconditional transfers.
+
+#### Request Format
+
+POST a JSON [Transfer resource][] to the `transfers` [URL][].
+
+**Metadata URL Name:** `transfers`
+
+##### Body Parameters
+
+The message body should be a JSON [Transfer resource][].
+
+#### Response Format
+
+A successful result uses the HTTP response code **201 Created** and contains a JSON body with the [Transfer resource][] as saved.
+
+#### Example
+
+Request:
+
+```json
+POST /transfers
+Content-Type: application/json
+
+{
+  "client_id": "3a2a1d9e-8640-4d2d-b06c-84f2cd613204",
+  "ledger": "http://usd-ledger.example",
+  "debit_account": "http://usd-ledger.example/accounts/alice",
+  "credit_account": "http://usd-ledger.example/accounts/bob",
+  "amount": "50",
+  "execution_condition": "cc:0:3:8ZdpKBDUV-KX_OnFZTsCWB_5mlCFI3DynX5f5H2dN-Y:2",
+  "expires_at": "2016-10-25T08:41:59.795Z"
+}
+```
+
+Response:
+
+```json
+HTTP/1.1 201 Created
+
+{
+  "id": "http://usd-ledger.example/transfers/3a2a1d9e-8640-4d2d-b06c-84f2cd613204",
+  "client_id": "3a2a1d9e-8640-4d2d-b06c-84f2cd613204",
+  "ledger": "http://usd-ledger.example",
+  "debit_account": "http://usd-ledger.example/accounts/alice",
+  "credit_account": "http://usd-ledger.example/accounts/bob",
+  "amount": "50",
+  "execution_condition": "cc:0:3:8ZdpKBDUV-KX_OnFZTsCWB_5mlCFI3DynX5f5H2dN-Y:2",
+  "expires_at": "2016-10-25T08:41:59.795Z",
+  "state": "prepared",
+  "timeline": {
+    "prepared_at": "2015-06-16T00:00:01.000Z"
+  }
+}
+```
+
+#### Errors
+
+- [InsufficientFundsError][] - The `debit_account` would go below its `minimum_allowed_balance` if this transfer were executed
+- [UnprocessableEntityError][] - The request is formatted properly but contains an otherwise-unspecified semantic problem.
+- [AlreadyExistsError][] - The `client_id` supplied is already used by another transfer.
+- [InvalidUriParameterError][] - One of the [URL][] or [UUID][] parameters was not formatted properly.
+- [InvalidBodyError][] - The request body was not properly-formatted JSON, or did not match the Content-Type provided.
```

Comment:
I'm not sure. I thought that would be an UnprocessableEntityError, but maybe not.